daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Command.scala

```diff
@@ -21,7 +21,9 @@ object Command {
 
   final case class Exercise(
       templateId: Identifier,
-      contractId: SContractId,
+      // This is an SExpr because we need to refer to the relative contract id
+      // of a CreateAndExerciseCommand with an SEVar reference.
+      contractId: SExpr,
```

Comment:
I do not really like the `SExpr`. We lost to think with that :

1. the (daml-lf) type of `contractId`
2. the fact that `contractId` cannot be evaluated (by the engine) as part of the command execution. 

I would prefer to have one new command  `CreateAndExerciseCommand` that just do not need the variable.
Or at least to use a special case class for contractId.
``
sealed abstract class ContractIdOrBinder
object ContractIdOrBinder{
  case object Binder extends ContractIdOrBinder
  case class ContractId (cid: SContractId)
}
```

Reply:
Done.